apps/src/lib/levelbuilder/lesson-editor/UploadImageForm.jsx

```diff
@@ -0,0 +1,114 @@
+import React, {useState} from 'react';
+
+import Button from '@cdo/apps/templates/Button';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+import i18n from '@cdo/locale';
+import styles from './uploadImage.module.scss';
+import classnames from 'classnames';
+
+export default function UploadImageForm() {
+  const [imgUrl, setImgUrl] = useState(undefined);
+  const [error, setError] = useState(undefined);
+  const [isUploading, setIsUploading] = useState(false);
+  const [formDataForImage, setFormDataForImage] = useState(undefined);
+  const [tempImageUrl, setTempImageUrl] = useState(undefined);
+
+  const resetState = () => {
+    setImgUrl(undefined);
+    setError(undefined);
+  };
+
+  const handleChange = e => {
+    if (!e.target.files[0]) {
+      resetState();
+      return;
+    }
+    resetState();
+
+    // assemble upload data
+    const formData = new FormData();
+    formData.append('file', e.target.files[0]);
+    setFormDataForImage(formData);
+    setTempImageUrl(URL.createObjectURL(e.target.files[0]));
+  };
+
+  const saveImageToS3 = () => {
+    setIsUploading(true);
+    // POST
+    const csrfContainer = document.querySelector('meta[name="csrf-token"]');
+    fetch('/level_assets/upload', {
+      method: 'post',
+      body: formDataForImage,
+      headers: {
+        'X-CSRF-Token': csrfContainer && csrfContainer.content,
+      },
+    })
+      .then(response => response.json())
+      .then(handleResult)
+      .catch(err => {
+        setError(err);
+        setIsUploading(false);
+      });
+  };
+
+  const handleResult = result => {
+    if (result && result.newAssetUrl) {
+      setImgUrl(result.newAssetUrl);
+    } else if (result && result.message) {
+      setError(result.message);
+    } else {
+      setError(result);
+    }
+    setIsUploading(false);
+  };
+
+  return (
+    <div className={styles.topContainer}>
+      <h2>{i18n.uploadImage()}</h2>
+      {
+        // TODO: A11y279 (https://codedotorg.atlassian.net/browse/A11Y-279)
```

Comment:
Could we just add some alt-text here that makes sense? I assume that this was added because someone didn't want to go through lots of different images to add alt-text, but since we are just adding one, I think we can add it.

Reply:
Maybe something like `uploaded image preview`

Reply:
Re: link to this page.  Yes, that is one of the follow ups from this ticket: `(Extension for “Image Upload page) Add the images link to the helpful links page`.  I documented this in your google doc as well and will be an easy fast follow.

Reply:
As I understand this comment (which was borrowed from a component that is used to upload images on a specific level) I think this might be a much bigger to do... right now this button just uploads it to S3 - there is no alt text stored on images currently in S3.  

So, I am not sure it is really easy to add alt-text when the image is uploaded to S3.  Instead it seems like alt-text should be added when a levelbuilder decides to use a level OR we need to add a table that includes URLs for images and alt text for those associated levels... again, feels like a much bigger project, unless I misunderstood the comment.

Reply:
Agreed that would be a much bigger ask. These comments were added as part of [this](https://github.com/code-dot-org/code-dot-org/pull/55445/files) PR and are not all related to not storing alt text. Some seem to be just because Kelby didn't know what alt-text to put and would have needed further exploration.

What do you think about adding a generic alt-text. Something like `uploaded image preview`?

Reply:
I see, in that case - yes!  I added this in.